src/core/tokenizer.c

```diff
@@ -195,3 +197,78 @@ tokenize(filter_pipeline_t *fp, const char *text, size_t text_len __unused)
 	free(content);
 	return NULL;
 }
+
+
+
+/*
+ * tokenize: split the text into tokens separated by a whitespace
+ * and pass the token through the filter pipeline.
+ */
+tokenset_t *
+tokenize2(filter_pipeline_t *fp, const char *text, size_t text_len __unused)
+{
+
+	UBreakIterator *it_token;
+	UChar          utext[text_len + 1]; 
+	UErrorCode     ec = U_ZERO_ERROR;
+	int32_t end;
+	int32_t start;
+	char *content;
+	tokenset_t *tset;
+	char val[10];
+
+	if ((tset = tokenset_create()) == NULL) {
+		return NULL;
+	}
+	if ((content = strdup(text)) == NULL) {
+		goto err;
+	}
+
+	u_uastrcpy(utext, text);
+	it_token = ubrk_open(UBRK_WORD, "en_us", utext, u_strlen(utext), &ec);
```

Comment:
- Optimization: `u_strlen()` is O(n) operation, so we don't want it. You already have the text size as a parameter (and which you would use to allocate your buffer).
- We have params (passed as nxs_params_t) in the caller which we should pass here. We store "lang" in them which represents the language which you can use here, instead of hardcoding the "en_us". The language codes we use are two-letter ISO 639-1 codes. However, ICU has its own "locale" codes which I think are a superset.